pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -198,6 +197,116 @@ var _ = Describe("podNIC", func() {
 			})
 		})
 	})
+	type stateRetrievalTestCase struct {
+		storedState   *cache.PodIfaceState
+		expectedState cache.PodIfaceState
+	}
+	statePointer := func(state cache.PodIfaceState) *cache.PodIfaceState {
+		return &state
+	}
+	DescribeTable("state is retrieved", func(c stateRetrievalTestCase) {
+		vmi := newVMIMasqueradeInterface("testnamespace", "testVmName", masqueradeCidr, masqueradeIpv6Cidr)
+		podnic, err := newPhase1PodNICWithMocks(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		if c.storedState != nil {
+			podnic.setState(*c.storedState)
+			c.expectedState = *c.storedState
+		}
+		obtainedState, err := podnic.state()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(obtainedState).To(Equal(c.expectedState))
+	},
+		Entry("when there is no pod interface cache should return PodIfaceNetworkPreparationPending", stateRetrievalTestCase{
+			storedState:   nil,
+			expectedState: cache.PodIfaceNetworkPreparationPending,
+		}),
+		Entry("when PodIfaceNetworkPreparationStarted is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationStarted),
+		}),
+		Entry("when PodIfaceNetworkPreparationPending is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationPending),
+		}),
+		Entry("when PodIfaceNetworkPreparationFinished is stored should return it", stateRetrievalTestCase{
+			storedState: statePointer(cache.PodIfaceNetworkPreparationFinished),
+		}),
+	)
+	Context("state transition", func() {
+		When("state is PodIfaceNetworkPreparationPending", func() {
+			var (
+				podnic *podNIC
+			)
+			BeforeEach(func() {
+				var err error
+				vmi := newVMIBridgeInterface("testnamespace", "testVmName")
+				podnic, err = newPhase1PodNICWithMocks(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				pid := 1
+				podnic.launcherPID = &pid
+				podnic.setState(cache.PodIfaceNetworkPreparationPending)
+				mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return("1.2.3.4", "169.254.0.0", nil)
+				mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil)
+				mockPodNetworkConfigurator.EXPECT().DiscoverPodNetworkInterface(podnic.podInterfaceName)
+				mockPodNetworkConfigurator.EXPECT().GenerateDomainIfaceSpec().Return(api.Interface{})
+				mockPodNetworkConfigurator.EXPECT().GenerateNonRecoverableDHCPConfig().Return(&cache.DHCPConfig{})
```

Comment:
nit: I would prefer if you used a different block for these

Reply:
One of them depend on podnic `mockPodNetworkConfigurator.EXPECT().DiscoverPodNetworkInterface(podnic.podInterfaceName)` and I am not sure if BeforeEach blocks get executed in the order they are declared.

Reply:
they are.